### DISCUSSION TITLE
Example C++ inferrence

### DISCUSSION THREAD

**Discussion by pfeatherstone** [google/flax#1673] [Q&A] [answered]
Is there a way to infer Flax models in C++ using a minimal library?
If so, could you provide an example?

**Comment by gnecula** [comment, marked as answer]
One method to try is to convert the inference function of your model to a TF function [using jax2tf](https://github.com/google/jax/tree/main/jax/experimental/jax2tf). Then you can use standard TensorFlow tooling for loading the SavedModel and running it from C++ (some pointers from [TF SavedModel documentation](https://www.tensorflow.org/guide/saved_model#load_a_savedmodel_in_c)).

**Reply by marcvanzee** [reply]
For deploying Flax models on mobile you could try using jax2tf, and then use the TFLite converter to convert your model to a TFLite model. This is still experimental, and not all ops will work, but there is an example in the JAX repo that you could use as a starting point: https://github.com/google/jax/tree/main/jax/experimental/jax2tf/examples/tflite

**Comment by pfeatherstone** [comment]
Is this the standard method for deploying Flax models? For example, deploying to a mobile device? Or is Flax not really designed for deployment, more for experimentation?
